src/bin/gpfusion/gpbridgeapi.c

```diff
@@ -536,3 +536,8 @@ void free_token_resources(PxfInputData *inputData)
 
 	pfree(inputData->token);
 }
+
+void free_dfs_address()
+{
+	free(dfs_address);
+}
```

Comment:
I checked free_dfs_address is called by gpbridge_export_start and gpbridge_import_start, do you mean this two funtions are only called in the secure mode? If not, you are freeing NULL pointer.